### PR TITLE
added my stuff

### PR DESCRIPTION
**Pull request by MyK00L**

### CODE REVIEW

**Review by eutampieri — CHANGES_REQUESTED** [review]
Interfaces should be prefixed with I, like IInterfaceName

File: Chelli/ITileMap.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+/// <summary>ITileMap represents a map for the game.</summary>
+public interface ITileMap
+{
+	/// <returns>the height of the map in tiles.</returns>
+	int Height();
+
+	/// <returns>the width of the map in tiles.</returns>
+	int Width();
```
Comment (eutampieri) [inline Chelli/ITileMap.cs]: why not using a property?

File: Chelli/ITileMap.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+	/// <param name="x">column.</param>
+	/// <param name="y">row.</param>
+	/// <returns>the Tile at column x and row y in the map.</returns>
+	Tile At(int x, int y);
```
Comment (eutampieri) [inline Chelli/ITileMap.cs]: There must be a IEnumerable way to do this

File: Chelli/Tile.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+public enum Tile
+{
+	/// <summary>A void tile, which contains nothing.</summary>
+	VOID,
+	/// <summary>A wall tile, which forms a wall.</summary>
+	WALL,
+	/// <summary>A floor tile, which makes floors.</summary>
+	FLOOR,
+	/// <summary>A stair tile, which will form flight of stairs.</summary>
+	STAIRS,
```
Comment (eutampieri) [inline Chelli/Tile.cs]: I think they should be like `Void`, `Wall`, etc.

File: Chelli/TileMapFactoryImpl.cs [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+			else
+			{
+				if (x < b.x)
+				{
+					x++;
+				}
+				else
+				{
+					x--;
+				}
```
Comment (eutampieri) [inline Chelli/TileMapFactoryImpl.cs:82]: wow love u c#

Comment (MyK00L) [reply Chelli/TileMapFactoryImpl.cs:82]: ikr

**Review by eutampieri — APPROVED** [review]

**Pull request merged**